File: C++/A/950A-lefthanders.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int l, r, a;
    cin >> l >> r >> a;
    int min = ((l < r) ? l : r);
    int max = ((l > r) ? l : r);

    int diff = max - min;

    if (diff == 0)
    {
        min += a / 2;
        max += a / 2;
    }
    else
    {
        if (a >= diff)
        {
            min += diff;
            a -= diff;
            min += a / 2;
            max += a / 2;
        }
        else
        {
            min += a;
        }
    }

    cout << 2 * min << "\n";
}

File: Python/A/950A-lefthanders.py
l, r, a = map(int, input().split())

minimum = min(l, r)
maximum = max(l, r)

difference = maximum - minimum

if difference == 0:
    minimum += a // 2
    maximum += a // 2
else:
    if a >= difference:
        minimum += difference
        a -= difference
        minimum += a // 2
        maximum += a // 2
    else:
        minimum += a

print(2 * minimum)
